sdk/identity/Azure.Identity/src/InteractiveBrowserCredential.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Microsoft.Identity.Client;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Identity
+{
+    /// <summary>
+    /// A <see cref="TokenCredential"/> implementation which launches the system default browser to interactively authenticate a user, and obtain an access token.
+    /// The browser will only be launched to authenticate the user once, then will silently aquire access tokens through the users refresh token as long as it's valid.
+    /// </summary>
+    public class InteractiveBrowserCredential : TokenCredential
+    {
+        private IPublicClientApplication _pubApp = null;
+        private IAccount _account = null;
+        private IdentityClientOptions _options;
+        private string _clientId;
+
+        /// <summary>
+        /// Creates a new InteractiveBrowserCredential which will authenticate users with the specified application.
+        /// </summary>
+        /// <param name="clientId">The client id of the application to which the users will authenticate.</param>
+        /// TODO: need to link to info on how the application has to be created to authenticate users, for multiple applications
+        public InteractiveBrowserCredential(string clientId)
+            : this (clientId, null)
+        {
+
+        }
+
+        /// <summary>
+        /// Creates a new DeviceCodeCredential with the specifeid options, which will authenticate users with the specified application.
+        /// </summary>
+        /// <param name="clientId">The client id of the application to which the users will authenticate</param>
+        /// TODO: need to link to info on how the application has to be created to authenticate users, for multiple applications
+        /// <param name="options">The client options for the newly created DeviceCodeCredential</param>
+        public InteractiveBrowserCredential(string clientId, IdentityClientOptions options)
+        {
+            _clientId = clientId ?? throw new ArgumentNullException(nameof(clientId));
+
+            _options = options ??= new IdentityClientOptions();
+
+            var pipeline = HttpPipelineBuilder.Build(_options, bufferResponse: true);
+
+            _pubApp = PublicClientApplicationBuilder.Create(_clientId).WithHttpClientFactory(new HttpPipelineClientFactory(pipeline)).WithRedirectUri("http://localhost").Build();
```

Comment:
Would we have problems with trying to bind to port 80? It's often taken.

Reply:
Seems like it's required by MSAL: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/MSAL.NET-uses-web-browser. But I'm also curious about identity team's decision behind it.

Reply:
Also on *nix machines you'll need root to open ports < 1024.

Reply:
Created https://github.com/Azure/azure-sdk-for-net/issues/7086 to address this.